File: src/main/java/de/kilobyte22/voc/api/network/Node.java
package de.kilobyte22.voc.api.network;

import de.kilobyte22.voc.api.Persistable;

public interface Node extends Persistable {

    Environment getHost();

    Visibility getVisibility();

    String getAddress();

    String getNetwork();

    boolean canBeReachedFrom(Node node);

    Iterable<Node> getReachableNodes();

    void sendToAddress(String address, String name, Object... data);

    void sendToReachable(String name, Object... data);

    void sendToVisible(String name, Object... data);

}

File: src/main/scala/de/kilobyte22/voc/api/network/Node.scala
package de.kilobyte22.voc.api.network

import de.kilobyte22.voc.api.Persistable

abstract trait Node extends Persistable {
  def host: Environment

  def visibility: Visibility

  def address: String

  def network: Network

  def canBeReachedFrom(node: Node): Boolean

  def reachableNodes: Iterable[Node]

  def sendToAddress(address: String, name: String, data: AnyRef*)

  def sendToReachable(name: String, data: AnyRef*)

  def sendToVisible(name: String, data: AnyRef*)
}